definition_tooling/validator/core.py

```diff
@@ -0,0 +1,114 @@
+import abc
+import json
+from pathlib import Path
+from typing import Union
+
+from definition_tooling.api_errors import DATA_PRODUCT_ERRORS
+from definition_tooling.validator import errors as err
+
+
+class BaseValidator:
+    def __init__(self, path: Union[str, Path]):
+        self.path = Path(path)
+
+    def validate(self):
+        try:
+            spec = json.loads(self.path.read_text(encoding="utf8"))
+        except json.JSONDecodeError:
+            raise err.InvalidJSON(f"Incorrect JSON: {self.path}")
+        except Exception as e:
+            raise err.ValidatorError(f"Failed to validate {self.path}: {e}")
+        self.validate_spec(spec)
+
+    @abc.abstractmethod
+    def validate_spec(self, spec: dict):
+        raise NotImplementedError
+
+
+class DefaultValidator(BaseValidator):
```

Comment:
I can't find anything that uses this `DefaultValidator` anywhere. The cli only uses the `DefinitionValidator`. What do you say, would it make sense to flatten it down to just have the `DefinitionValidator` and no `BaseValidator` at the current time, or should be consider adding support to this so that you could specify which validator(s) to run?

Reply:
Yes, it was the idea, yet we don't have any plans for that, so it indeed looks like unnecessary code